### DISCUSSION TITLE
Pagination

### DISCUSSION THREAD

**Discussion by robertjmccarthy** [ministryofjustice/moj-frontend#708] [RELEASED: Components, pages and patterns]
The [pagination component](https://design-patterns.service.justice.gov.uk/components/pagination/) is live in the MoJ Design System.

Use this discussion to suggest changes and improvements to this component.

**Comment by robertjmccarthy** [comment]
@mary-bowden flagged an issue with the pagination component where it shows an odd content label for '_Showing 10 to 20 of 30 pages_'.

Suggest changing '_pages_' to '_results_'.

@helennickols could you review this suggestion please?

---

<img width="781" alt="Screenshot 2024-09-10 at 21 10 13" src="https://github.com/user-attachments/assets/68e43938-f4d4-4254-bc18-5f263fa9822b">

**Reply by helennickols** [reply]
Yes definitely, I'll do this today.

**Reply by helennickols** [reply]
This is complete

**Comment by andreea-moj** [comment]
Hello :) I wanted to share two suggestions that might help improve this pagination component:

1. total results numbers - in the example provided, the message says “Showing 11 to 20 of 30 results,” but there are 5+ pages available. This suggests a total closer to 70 results. Would it make sense to update the example to something like “Showing 11 to 20 of 60 results” to better align with the pagination?

2. first and last page view - the current example does a great job showing a 'mid-range' pagination state. To make it more flexible and ready-to-use, would it be worth following the GOV.UK pattern for pagination? They include "First" and "Last" page links ([example here](https://design-system.service.gov.uk/components/pagination/)) which can be helpful for accessibility and usability.

I'm happy to share more code snippets and a link to a prototype if that would be useful! 😊

`		<nav class="moj-pagination" aria-label="Pagination navigation">

			<ul class="moj-pagination__list">
			 
		  
			  <li class="moj-pagination__item moj-pagination__item--active" aria-label="Page 1 of 5" aria-current="page">1</li>

			  <li class="moj-pagination__item"><a class="moj-pagination__link" href="/page=1" aria-label="Page 2 of 5">2</a></li>
		  
			  <li class="moj-pagination__item"><a class="moj-pagination__link" href="/page=3" aria-label="Page 3 of 5">3</a></li>
		  
			  <li class="moj-pagination__item moj-pagination__item--dots">…</li>
		  
			  <li class="moj-pagination__item"><a class="moj-pagination__link" href="/page=5" aria-label="Page 5 of 5">5</a></li>
		  
			  <li class="moj-pagination__item  moj-pagination__item--next">
				<a class="moj-pagination__link" href="">Next<span class="govuk-visually-hidden"> page</span></a>
			  </li>
			</ul>
			<p class="moj-pagination__results">Showing <b>1</b> to <b>10</b> of <b>60</b> results</p>
		  </nav>`

<img width="1470" height="87" alt="image" src="https://github.com/user-attachments/assets/97e8abaf-04c9-4bcd-ad7d-711b5313bdb3" />

**Reply by Whersi** [reply]
Hi Andrea,

Thank you for sharing your comments on the Pagination component.

For your first point about improving the example in the documentation, I’ll raise an issue for the team to look at and update if possible within this cycle.

Regarding your second suggestion — the pagination component is next in line for a full review, so your idea about including different examples is really helpful and will be considered. Would you be happy for us to note your name in case you'd like to take part in our upcoming discovery work on it in September?

**Comment by andreea-moj** [comment]
Hello and thanks for your reply. Yes - happy to help and you can use my name. You are right about being in line for a full review, hence thinking the comment will be useful 😊

Thank you,
Andreea.
From: Whersi ***@***.***>
Date: Tuesday, 5 August 2025 at 13:39
To: ministryofjustice/moj-frontend ***@***.***>
Cc: Vlad, Andreea ***@***.***>, Comment ***@***.***>
Subject: Re: [ministryofjustice/moj-frontend] Pagination (Discussion #708)


Hi Andrea,

Thank you for sharing your comments on the Pagination component.

For your first point about improving the example in the documentation, I’ll raise an issue for the team to look at and update if possible within this cycle.

Regarding your second suggestion — the pagination component is next in line for a full review, so your idea about including different examples is really helpful and will be considered. Would you be happy for us to note your name in case you'd like to take part in our upcoming discovery work on it in September?

—
Reply to this email directly, view it on GitHub<https://github.com/ministryofjustice/moj-frontend/discussions/708#discussioncomment-14005819>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/BJ67VQ3FN64TLY4DWU2VV6T3MCQYHAVCNFSM6AAAAACC5E24CSVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTIMBQGU4DCOI>.
You are receiving this because you commented.

________________________________
This e-mail and any attachments is intended only for the attention of the addressee(s). Its unauthorised use, disclosure, storage or copying is not permitted. If you are not the intended recipient, please destroy all copies and inform the sender by return e-mail. Internet e-mail is not a secure medium. Any reply to this message could be intercepted and read by someone else. Please bear that in mind when deciding whether to send material in response to this message by e-mail. This e-mail (whether you are the sender or the recipient) may be monitored, recorded and retained by the Ministry of Justice. Monitoring / blocking software may be used, and e-mail content may be read at any time. You have a responsibility to ensure laws are not broken when composing or forwarding e-mails and their contents.

**Comment by lalexander12** [comment]
Thinking about the 'Showing 1 to 10 of X results' bit and whether this should be invisible when one page of results?

I see that the whole component can be configured to be hidden when just 1 page of results. 

For example, on one page of 8 results, would it be preferable to keep the bit saying 'Showing 1 to 8 of 8 results.' It seems a little verbose. Should it be shortened to just say 8 results? Or removed completely.

And, ideally, it would be preferable to be specific about the type of item being counted. For example, 8 schools, 8 courts, 8 claims.

Some examples:

<img width="300" height="250" alt="Screenshot 2025-09-04 at 11 18 11" src="https://github.com/user-attachments/assets/79529274-22eb-4d0f-8e94-06494613366e" />

<img width="350" height="500" alt="Screenshot 2025-09-04 at 11 17 13" src="https://github.com/user-attachments/assets/ee57d8d1-5fb2-43db-adac-73124c1892dd" />
<img width="450" height="304" alt="Screenshot 2025-09-04 at 11 17 27" src="https://github.com/user-attachments/assets/aed83964-a7db-463a-bf21-0a0fd0d34e17" />

<img width="500" height="520" alt="Screenshot 2025-09-04 at 11 16 55" src="https://github.com/user-attachments/assets/1d20ccff-2de8-4ff0-9459-0518d8e848c5" />
